keras_tuner/engine/hyperparameters.py

```diff
@@ -557,6 +563,13 @@ def __init__(self):
         # Active values for this `Trial`.
         self.values = {}
 
+        # Lists of stacks of conditions used during `explore_space()`.
+        self._scopes_never_active = []
+        self._scopes_once_active = []
```

Comment:
You could move these 2 attributes to the `explore_space` method, they are never accessed before `explore_space` is called. They're somewhat surprising in the constructor